Snowplow/Internal/Tracker/SPTracker.m

```diff
@@ -620,17 +623,33 @@ - (void)addSelfDescribingPropertiesToPayload:(SPPayload *)payload event:(SPTrack
 /*
  This is needed because the campaign-attribution-enrichment (in the pipeline) is able to parse
  the `url` and `referrer` only if they are part of a PageView event.
- The PageView event is an atomic event but the DeepLinkReceived is a SelfDescribing event.
+ The PageView event is an atomic event but the DeepLinkReceived and ScreenView are SelfDescribing events.
  For this reason we copy these two fields in the atomic fields in order to let the enrichment
  to process correctly the fields even if the event is not a PageView and it's a SelfDescribing event.
  This is a hack that should be removed once the atomic event table is dismissed and all the events
  will be SelfDescribing.
  */
-- (void)workaroundForCampaignAttributionEnrichment:(SPPayload *)payload event:(SPTrackerEvent *)event {
+- (void)workaroundForCampaignAttributionEnrichment:(SPPayload *)payload event:(SPTrackerEvent *)event contexts:(NSMutableArray<SPSelfDescribingJson *> *)contexts {
+    NSString *url = nil;
+    NSString *referrer = nil;
+    
     if ([event.schema isEqualToString:kSPDeepLinkReceivedSchema]) {
-        NSString *url = (NSString *)[event.payload objectForKey:kSPDeepLinkReceivedParamUrl];
-        NSString *referrer = (NSString *)[event.payload objectForKey:kSPDeepLinkReceivedParamReferrer];
+        url = (NSString *)[event.payload objectForKey:kSPDeepLinkReceivedParamUrl];
+        referrer = (NSString *)[event.payload objectForKey:kSPDeepLinkReceivedParamReferrer];
+    } else if ([event.schema isEqualToString:kSPScreenViewSchema]) {
+        for (SPSelfDescribingJson *entity in contexts) {
+            if ([[entity schema] isEqualToString:kSPDeepLinkSchema]) {
+                NSDictionary *data = (NSDictionary *)[entity data];
+                url = (NSString *)[data valueForKey:kSPDeepLinkReceivedParamUrl];
+                referrer = (NSString *)[data valueForKey:kSPDeepLinkParamReferrer];
```

Comment:
Maybe we can add here `break;` so it doesn't continue to compare strings after execution of this block.